src/main/java/XbeLoader/XbeImageHeader.java

```diff
@@ -22,37 +22,95 @@
 import ghidra.util.task.TaskMonitor;
 
 public class XbeImageHeader {
-	public long   magic;
-	public byte[] signature;
-	public long   baseAddr;
-	public long   headersSize;
-	public long   imageSize;
-	public long   imageHeaderSize;
-	public long   timestamp;
-	public long   certificateAddr;
-	public long   sectionCount;
-	public long   sectionHeadersAddr;
-	public long   initFlags;
-	public long   entryAddr;
-	public long   tlsAddr;
-	public long   peStackCommit;
-	public long   peHeapReserve;
-	public long   peHeapCommit;
-	public long   peBaseAddr;
-	public long   peImageSize;
-	public long   peChecksum;
-	public long   peTimestamp;
-	public long   debugPathnameAddr;
-	public long   debugFilenameAddr;
-	public long   debugUnicodeFilenameAddr;
-	public long   kernThunkAddr;
-	public long   importDirAddr;
-	public long   libVersionsCount;
-	public long   libVersionsAddr;
-	public long   kernLibVersionAddr;
-	public long   xapiLibVersionAddr;
-	public long   logoAddr;
-	public long   logoSize;
+	public enum XbeType {
+		RETAIL(0),
+		DEBUG(1),
+		CHIHIRO(2);
+
+		private int value;
+
+		private XbeType(int setValue) {
+			value = setValue;
+		}
+
+		public int getValue() {
+			return value;
+		}
+	}
+
+	private long[] xorEntryPoint = {
+		0xA8FC57ABL, // Retail
+		0x94859D4BL, // Debug
+		0x40B5C16EL  // Chihiro
+	};
+
+	private long[] xorKernelThunk = {
+		0x5B6D40B6L, // Retail
+		0xEFB1F152L, // Debug
+		0x2290059DL  // Chihiro
+	};
+
+	public long    magic;
+	public byte[]  signature;
+	public long    baseAddr;
+	public long    headersSize;
+	public long    imageSize;
+	public long    imageHeaderSize;
+	public long    timestamp;
+	public long    certificateAddr;
+	public long    sectionCount;
+	public long    sectionHeadersAddr;
+	public long    initFlags;
+	public long    entryAddr;
+	public long    tlsAddr;
+	public long    peStackCommit;
+	public long    peHeapReserve;
+	public long    peHeapCommit;
+	public long    peBaseAddr;
+	public long    peImageSize;
+	public long    peChecksum;
+	public long    peTimestamp;
+	public long    debugPathnameAddr;
+	public long    debugFilenameAddr;
+	public long    debugUnicodeFilenameAddr;
+	public long    kernThunkAddr;
+	public long    importDirAddr;
+	public long    libVersionsCount;
+	public long    libVersionsAddr;
+	public long    kernLibVersionAddr;
+	public long    xapiLibVersionAddr;
+	public long    logoAddr;
+	public long    logoSize;
+	public XbeType type;
+
+	// ported from Dxbx's XbeExplorer
+	private XbeType GetXbeType() {
```

Comment:
nit: I suggest adding a comment here which makes clear that this isn't how an XBE loader normally works. This is merely a heuristic to find the constants that the appropriate kernel for such an XBE would use